guides/tutorial/subscriptions.md

```diff
@@ -0,0 +1,144 @@
+# Subscriptions
+
+When the need arises for near realtime data GraphQL provides subscriptions. We want to support subscriptions that look like
+
+
+```graphql
+subscription{
+  newPost {
+    id
+    name
+  }
+}
+```
+
+Since we had already setup mutations to handle creation of posts we can use that as the event we want to subscribe to. In order to achieve this we have to do a little bit of set up
+
+
+Let's start by adding `absinthe_phoenix` as a dependency
+
+In `mix.exs`
+
+```elixir
+defp deps do
+  [
+    {:absinthe_phoenix, "~> 1.4.0"}
+    << other deps >>
+  ]
+```
+
+Then we need to add a supervisor to run some processes for the to handle result broadcasts
+
+In `lib/blog/application.ex`:
+
+```elixir
+  children = [
+    <<other supervisors>>
+
+    supervisor(Absinthe.Subscription, [BlogWeb.Endpoint])
+  ]
+```
+
+The lets add a configuration to the phoenix endpoint so it can provide some callbacks Absinthe expects, please note while this guide uses phoenix Absinthes support for Subscriptions is good enough to be used without websockets even without a browser.
+
+In `lib/blog_web/endpoint.ex`:
+
+
+```elixir
+defmodule BlogWeb.Endpoint do
+  use Phoenix.Endpoint, otp_app: :blog
+  use Absinthe.Phoenix.Endpoint
+
+  << rest of the file>>
+```
+
+The pubsub stuff is now set up lets configure our sockets
+
+In `lib/blog_web/channels/user_socket.ex`
+
+``` elixir
+defmodule BlogWeb.UserSocket do
+  use Phoenix.Socket
+  use Absinthe.Phoenix.Socket, schema: BlogWeb.Schema
+
+  << rest of file>>
+```
+
+Lets not configure GraphQL to use this socket.
+
+In `lib/blog_web/router.ex` :
+
+```elixir
+defmodule BlogWeb.Router do
+  use BlogWeb, :router
+
+  pipeline :api do
+    plug :accepts, ["json"]
+    plug BlogWeb.Context
+  end
+
+  scope "/api" do
+    pipe_through :api
+
+    forward "/graphiql", Absinthe.Plug.GraphiQL,
+      schema: BlogWeb.Schema,
+      socket: BlogWeb.UserSocket
+
+    forward "/", Absinthe.Plug,
+      schema: BlogWeb.Schema,
+      socket: BlogWeb.UserSocket
```

Comment:
The socket isn't necessary here. The socket is necessary for GraphiQL because GraphIQL runs javascript that connects via the socket. The plain plug route doesn't do anything with the socket.

Reply:
Is that so?